core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1905,6 +1905,13 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
 
   @nowarn("cat=deprecation")
   private def validateValues(): Unit = {
+    val nodeIdValue = values.get(KafkaConfig.NodeIdProp).asInstanceOf[Int]
+    if (nodeIdValue >= 0) {
+      val brokerIdValue = values.get(KafkaConfig.BrokerIdProp).asInstanceOf[Int]
```

Comment:
```suggestion
    val nodeIdValue = getInt(KafkaConfig.NodeIdProp)
    if (nodeIdValue >= 0) {
      val brokerIdValue = getInt(KafkaConfig.BrokerIdProp)
```
For consistency, this type prefers to use `getInt` which does the casting.

Reply:
> prefers to use getInt

God catch -- fixed.